sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -416,11 +419,17 @@ abstract class UnixTime extends BinaryExpression with ExpectsInputTypes {
         }
       case StringType =>
         val sdf = classOf[SimpleDateFormat].getName
+        val formatter = ctx.freshName("sdf")
+        ctx.addMutableState(sdf, formatter, s"""$formatter = null;""")
         nullSafeCodeGen(ctx, ev, (string, format) => {
           s"""
             try {
+              if ($formatter == null ||
+                  !$formatter.toPattern().equals("$format.toString()")) {
```

Comment:
So, this raises an interesting point ... the SimpleDateFormat you're caching is a reference used across logically different code branches that need different formats. Are we sure this is a win? because every time a different format is needed you have to switch anyway. I might just be paranoid